gpii/node_modules/flowManager/src/DefaultSettingsLoader.js

```diff
@@ -18,34 +19,73 @@ var fluid = fluid || require("infusion"),
     path = require("path"),
     JSON5 = require("json5");
 
+// The default settings loader reads default settings thru these steps:
+// 1. Fetch default settings from the remote URL;
+// 2. If the remote fetch fails, read default settings from the local system settings directory. In the case that the
+// default settings file doesn't exists in the system settings directory, copy it from the code base to the settings
+// directory.
+// If the default settings file is not found at both places, return undefined.
 fluid.defaults("gpii.defaultSettingsLoader", {
     gradeNames: ["fluid.component"],
     // The path to the file that has all QSS supported settings.
     defaultSettingsInCodeBase: "%gpii-universal/testData/defaultSettings/defaultSettings.json5",
+    // The url to the remote default settings file. Must be provided by integrators.
+    defaultSettingsUrl: null,
     members: {
         gpiiSettingsDir: "@expand:{settingsDir}.getGpiiSettingsDir()",
-        // It points to the default settings file in the settings directory. GPII should read from this file instead of defaultSettingsInCodeBase.
-        defaultSettingsFile: "@expand:gpii.defaultSettingsLoader.prepareSettingsFile({that}.options.defaultSettingsInCodeBase,{that}.gpiiSettingsDir)"
+        // Populated by "onCreate.prepareSettingsFile" listener. It points to the default settings file in
+        // the settings directory rather than defaultSettingsInCodeBase, the default settings file in the code base.
+        defaultSettingsFile: null
     },
     components: {
         settingsDir: {
             type: "gpii.settingsDir"
+        },
+        remoteDefaultSettingsDataSource: {
+            type: "kettle.dataSource.URL",
+            options: {
+                url: "{defaultSettingsLoader}.options.defaultSettingsUrl"
+            }
+        }
+    },
+    events: {
+        onGet: null
+    },
+    listeners: {
+        "onCreate.prepareSettingsFile": {
+            listener: "fluid.set",
+            args: ["{that}", ["defaultSettingsFile"], "@expand:gpii.defaultSettingsLoader.prepareSettingsFile({that}.options.defaultSettingsInCodeBase,{that}.gpiiSettingsDir)"]
+        },
+
+        // The promise chain to fetch default settings:
+        // 1. If defaultSettingsUrl is provided, the first attempt is to fetch from the remote url;
+        // 2. If the fetch remote fails, perform the first fallback to read from the local system settings directory;
+        // 3. If the read from the local fails, return undefined.
+        "onGet.getRemoteDefaultSettings": {
+            listener: "gpii.defaultSettingsLoader.getRemoteDefaultSettings",
+            args: ["{that}.options.defaultSettingsUrl", "{that}.remoteDefaultSettingsDataSource"]
```

Comment:
Some sort of priority would be helpful here, even if it is just "first" to indicate that there is a general expectation that this executes before some others